deepes/include/torch/es_agent.h

```diff
@@ -80,68 +96,79 @@ class Predictor{
+    if (is_eval) {
+      // predict with _model (without addding noise)
+      return _model->forward(x);
+    }
+    else {
+      // predict with _sampled_model (with adding noise)
+      return _sampled_model->forward(x);
+    }
   }
 
   bool update(std::vector<SamplingKey>& noisy_keys, std::vector<float>& noisy_rewards) {
     compute_centered_ranks(noisy_rewards);
-    float* noise = new float [_param_size];
-    float* neg_gradients = new float [_param_size];
-    memset(neg_gradients, 0, _param_size * sizeof(float));
+
+    memset(_neg_gradients, 0, _param_size * sizeof(float));
     for (int i = 0; i < noisy_keys.size(); ++i) {
       int key = noisy_keys[i].key(0);
       float reward = noisy_rewards[i];
-      bool success = _sampling_method->resampling(key, noise, _param_size);
-      for (int j = 0; j < _param_size; ++j) {
-        neg_gradients[j] += noise[j] * reward;
+      bool success = _sampling_method->resampling(key, _noise, _param_size);
+      for (int64_t j = 0; j < _param_size; ++j) {
+        _neg_gradients[j] += _noise[j] * reward;
       }
     }
-    for (int j = 0; j < _param_size; ++j) {
-      neg_gradients[j] /= -1.0 * noisy_keys.size();
+    for (int64_t j = 0; j < _param_size; ++j) {
+      _neg_gradients[j] /= -1.0 * noisy_keys.size();
     }
 
     //update
     auto params = _model->named_parameters();
-    int counter = 0;
+    int64_t counter = 0;
     for (auto& param: params) {
       torch::Tensor tensor = param.value().view({-1});
       auto tensor_a = tensor.accessor<float,1>();
-      _optimizer->update(tensor_a, neg_gradients+counter, tensor.size(0));
+      _optimizer->update(tensor_a, _neg_gradients+counter, tensor.size(0));
       counter += tensor.size(0);
     }
-    delete[] noise;
-    delete[] neg_gradients;
   }
 
   SamplingKey add_noise() {
     SamplingKey sampling_key;
     auto sampled_params = _sampled_model->named_parameters();
     auto params = _model->named_parameters();
-    float* noise = new float [_param_size];
-    int key = _sampling_method->sampling(noise, _param_size);
+    int key = _sampling_method->sampling(_noise, _param_size);
     sampling_key.add_key(key);
-    int counter = 0;
+    int64_t counter = 0;
     for (auto& param: sampled_params) {
       torch::Tensor sampled_tensor = param.value().view({-1});
       std::string param_name = param.key();
       torch::Tensor tensor = params.find(param_name)->view({-1});
       auto sampled_tensor_a = sampled_tensor.accessor<float,1>();
       auto tensor_a = tensor.accessor<float,1>();
-      for (int j = 0; j < tensor.size(0); ++j) {
-        sampled_tensor_a[j] = tensor_a[j] + noise[counter + j];
+      for (int64_t j = 0; j < tensor.size(0); ++j) {
+        sampled_tensor_a[j] = tensor_a[j] + _noise[counter + j];
       }
       counter += tensor.size(0);
     }
-    delete[] noise;
     return sampling_key;
   }
 
-  int param_size() {
+  int64_t param_size() {
```

Comment:
This function of paddle version is named as _caculate_param_size. Please update this function name to the same name.